components/carousel/script.js

```diff
@@ -0,0 +1,368 @@
+import { Components } from 'coherent-gameface-components';
+import template from './template.html';
+
+const components = new Components();
+const BaseComponent = components.BaseComponent;
+
+const CONTENT_WRAPPER_SELECTOR = '.guic-carousel-content-wrapper';
+const DOTS_CONTAINER_SELECTOR = '.guic-carousel-dots';
+const DOT_SELECTOR = '.guic-carousel-dot';
+const CAROUSEL_SELECTOR = '.guic-carousel';
+const NEXT_BTN_SELECTOR = '.guic-carousel-btn-next';
+const PREV_BTN_SELECTOR = '.guic-carousel-btn-prev';
+const NEXT_ARROW_SELECTOR = '.guic-carousel-right-arrow';
+const PREV_ARROW_SELECTOR = '.guic-carousel-left-arrow';
+
+/**
+ * Class description
+ */
+class Carousel extends BaseComponent {
+    /* eslint-disable-next-line require-jsdoc */
+    constructor() {
+        super();
+        this.template = template;
+        this.init = this.init.bind(this);
+
+        this._pageSize = 2;
+        this._navArrowStepSize = 1;
+        this.current = 0;
+        this.itemsDirection = 1;
+    }
+
+    /* eslint-disable-next-line require-jsdoc */
+    connectedCallback() {
+        components.loadResource(this)
+            .then(this.init)
+            .catch(err => console.error(err));
+    }
+
+    /* eslint-disable-next-line require-jsdoc */
+    init(data) {
+        this.setupTemplate(data, () => {
+            components.renderOnce(this);
+
+            this.contentWrapper = this.querySelector(CONTENT_WRAPPER_SELECTOR);
+            components.waitForFrames(() => {
+                this.itemWidth = this.items[0].getBoundingClientRect().width;
+                this.itemHeight = this.items[0].getBoundingClientRect().height;
+                this.resize();
+            });
+            this.style.visibility = 'visible';
+
+            this.createPaginationControls();
+            this.attachControlButtonsListeners();
+
+            this.shouldShowUpNavArrow();
+            this.shouldShowDownNavArrow();
+        });
+    }
+
+    /**
+     * Return the number of pages on the carousel.
+     * Calculated runtime. Returns 0 if there are no items.
+     */
+    get pages() {
+        return (this.items?.length / this.pageSize) || 0;
+    }
+
+    /**
+     * Return all carousel items.
+     */
+    get items() {
+        return this.querySelector(CONTENT_WRAPPER_SELECTOR).children;
+    }
+
+    /**
+     * Return the indexes of all items that are currently visible on the carousel.
+     * @returns {Array<string>}
+     */
+    get visibleItemsIndexes() {
+        let visibleElementsLength = this.pageSize;
+        let index = this.current;
+
+        const indexes = [];
+        while (visibleElementsLength) {
+            indexes.push(index);
+            index++;
+            visibleElementsLength--;
+        }
+
+        return indexes;
+    }

```

Comment:
Maybe we can do it simpler :D
Inspired by Misho

```suggestion
    get visibleItemsIndexes() {
        return Array.from({ length: this.pageSize }, (_, i) => i + this.current);
    }
```